src/executor/insert.rs

```diff
@@ -52,7 +52,12 @@ where
         if self.insert_struct.columns.is_empty() {
             TryInsertResult::Empty
         } else {
-            TryInsertResult::Inserted(self.insert_struct.exec(db).await)
+            let temp = self.insert_struct.exec(db).await;
+            if matches!(temp, Err(DbErr::RecordNotInserted)) {
+                TryInsertResult::Conflicted
+            } else {
+                TryInsertResult::Inserted(temp)
+            }
         }
```

Comment:
Coding Style: maybe a early return over nested `if ...`?

```rs
if self.insert_struct.columns.is_empty() {
    return TryInsertResult::Empty
} 
let temp = self.insert_struct.exec(db).await;
if matches!(temp, Err(DbErr::RecordNotInserted)) {
    TryInsertResult::Conflicted
} else {
    TryInsertResult::Inserted(temp)
}
```

Same applied to the methods below